client/src/pages/Messages/Conversation/Conversation.tsx

```diff
@@ -1,15 +1,70 @@
+import { useEffect, useState } from 'react';
+import { useAuth } from '../../../context/useAuthContext';
+import { ConversationProps } from './interface/Conversation';
+import { User } from '../../../interface/User';
+import { Message } from '../../../interface/Message';
+import fetchUser from '../../../helpers/APICalls/getUserById';
+import fetchLastMessage from '../../../helpers/APICalls/getMessagesByConvoId';
 import useStyles from './useStyles';
-import profilePic from '../../../Images/profile.png';
+import profilePicAvatar from '../../../Images/user.png';
 import { Typography, Grid, ListItem, Divider, ListItemAvatar, Avatar, ListItemText, Badge } from '@material-ui/core';
+import moment from 'moment';
 
-export default function Conversation(): JSX.Element {
+export default function Conversation({ convo, setConvo }: ConversationProps): JSX.Element {
   const classes = useStyles();
+  const { loggedInUser } = useAuth();
+  const participantId = convo.participants?.find((participant: string) => participant !== loggedInUser?.id);
+
+  const [participant, setParticipant] = useState<User | null>(null);
+  const [lastMessage, setLastMessage] = useState<Message | null>(null);
+
+  useEffect(() => {
+    let active = true;
+
+    if (participantId) {
+      const getParticipantInfo = async () => {
```

Comment:
When we are fetching a conversation, maybe we can populate the participant info and send last message details as well?
That would avoid the need for this api call and the one below it

Reply:
implemented the request with Context API